### PR TITLE
[MM-60678] Add confirmation modal when ending call for everyone

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Making it a little more difficult to accidentally end the call as it happened a couple of times already in our developers meeting.

#### Video

[-14-Town-Square---Calls-Mattermost.webm](https://github.com/user-attachments/assets/24d53503-a51a-4710-b62b-b020325e16ce)

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-60678

### CODE REVIEW

**Review by cpoile — APPROVED** [review]
Nice work, clean. 
I forget -- is there a slash command to end call for everyone?

**Comment by streamer45** [comment]
> Nice work, clean. I forget -- is there a slash command to end call for everyone?

@cpoile Yeah, that exited before the new functionality and comes with a prompt already.

That reminds me that I need to make some additions here, or this won't work from the global widget :facepalm:

**Comment by streamer45** [comment]
@cpoile Will need a second pass, given the above.

File: webapp/src/index.tsx [hunk 1]
```diff
@@ -321,6 +324,33 @@ export default class Plugin {
             document.getElementById('calls')?.remove();
         });
 
+        if (window.desktop) {
+            const widgetCh = new BroadcastChannel('calls_widget');
```
Comment (cpoile) [inline webapp/src/index.tsx:328]: Heh, cool.

**Review by cpoile — APPROVED** [review]
Awesome. But are we now getting into versioning and backwards compatibility issues?

**Comment by streamer45** [comment]
> Awesome. But are we now getting into versioning and backwards compatibility issues?

@cpoile This being a plugin-only implementation, I don't foresee compatibility issues. There's absolutely no Desktop change needed here.

**Comment by cpoile** [comment]
But as long as they have the version of desktop that supports this? (Or am I thinking of a different thing?)

**Comment by streamer45** [comment]
> But as long as they have the version of desktop that supports this? (Or am I thinking of a different thing?)

@cpoile No Desktop changes are needed for this to work, so there is no version requirement either. Check https://mattermost.atlassian.net/wiki/spaces/PE/pages/3013836814/Simplified+general-purpose+communication+between+JS+contexts+across+Desktop if you'd like to learn more about this approach.

**Comment by cpoile** [comment]
So this didn't even need desktop changes in the first place... I didn't fully grasp that part. Wow, nice.

**Comment by streamer45** [comment]
> So this didn't even need desktop changes in the first place... I didn't fully grasp that part. Wow, nice.

If we used this approach from the beginning, we could have simplified some stuff. Still we need some level of DesktopAPI to do stuff like focusing/switching servers.

**Pull request merged**